.github/workflows/integration_test.yaml

```diff
@@ -160,199 +175,134 @@ on:
+        with:
+          identifier: ${{ inputs.identifier }}
+          upload-image: ${{ inputs.upload-image }}
+          working-directory: ${{ inputs.working-directory }}
+
+  build:
+    name: Build ${{ matrix.build.type }} (${{ matrix.build.name }})
+    needs: [ plan ]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        build: ${{ fromJSON(needs.plan.outputs.plan).build }}
     steps:
       - uses: canonical/setup-lxd@v0.1.1
-
       - uses: actions/checkout@v4
-        if: inputs.charmcraft-repository != ''
+      - uses: canonical/operator-workflows/internal/build@better-build
+        id: build
         with:
-          repository: ${{ inputs.charmcraft-repository }}
-          ref: ${{ inputs.charmcraft-ref }}
-          path: ./charmcraft
-          fetch-depth: 0
-
-      - name: Get Charmcraft SHA
-        if: inputs.charmcraft-repository != ''
-        id: charmcraft-sha
-        working-directory: ./charmcraft
-        run: echo "sha=$(git rev-parse HEAD)" >> $GITHUB_OUTPUT
+          build-plan: ${{ toJSON(matrix.build) }}
+          charmcraft-channel: ${{ inputs.charmcraft-channel }}
+          charmcraft-ref: ${{ inputs.charmcraft-ref }}
+          charmcraft-repository: ${{ inputs.charmcraft-repository }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          rockcraft-channel: ${{ inputs.rockcraft-channel }}
+          rockcraft-ref: ${{ inputs.rockcraft-ref }}
+          rockcraft-repository: ${{ inputs.rockcraft-repository }}
 
-      - name: Restore Charmcraft Cache
-        if: inputs.charmcraft-repository != ''
-        id: restore-charmcraft
-        uses: actions/cache/restore@v4
+  plan-scan:
+    name: Plan Image Scanning
+    runs-on: ubuntu-latest
+    needs: [ plan, build ]
+    outputs:
+      scans: ${{ steps.plan-scan.outputs.scans }}
+    steps:
+      - uses: canonical/operator-workflows/internal/plan-scan@better-build
+        id: plan-scan
         with:
-          path: ./charmcraft*.snap
-          key: charmcraft-${{ steps.charmcraft-sha.outputs.sha }}
-
-      - name: Install Snapcraft
-        if: steps.restore-charmcraft.outputs.cache-hit != 'true' && inputs.charmcraft-repository != ''
-        run: sudo snap install snapcraft --classic
+          plan: ${{ needs.plan.outputs.plan }}
 
-      - name: Build Charmcraft
-        if: steps.restore-charmcraft.outputs.cache-hit != 'true' && inputs.charmcraft-repository != ''
-        working-directory: ./charmcraft
-        run: |
-          snapcraft --use-lxd
-          cp charmcraft*.snap ../
-
-      - name: Save Charmcraft Cache
-        uses: actions/cache/save@v4
-        if: steps.restore-charmcraft.outputs.cache-hit != 'true' && inputs.charmcraft-repository != ''
+  scan:
+    name: Scan Image (${{ matrix.scan.file }})
+    runs-on: ubuntu-latest
+    needs: [ plan-scan ]
+    strategy:
+      matrix:
+        scan: ${{ fromJSON(needs.plan-scan.outputs.scans) }}
+    steps:
+      - uses: actions/download-artifact@v4.17
```

Comment:
I believe you missed a dot here. The last version is 4.1.7

Reply:
Updated, thanks!